pkg/adapter/config/encryptionAdapter_test.go

```diff
@@ -0,0 +1,159 @@
+package config
+
+import (
+	"context"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func TestPublicKeyConfigEncryptionAdapter_Encrypt(t *testing.T) {
+	t.Run("can encrypt and decrypt", func(t *testing.T) {
+		// given
+		doguname := "testdogu"
+		namespace := "testingnamespace"
+		testValue := "value"
+		keyProviderStr := "pkcs1v15"
+		testPrivateKey := "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEA0D1M1hg9Tjwn8mU/FtnKG9bGaomVWBFTeMeZIy3vX0HgNr1H\nNFCt7yHEE2H+BZkK8nJ2BHBeMw8hsmndmGuGyZrtWe8pxbl1x79/d+kcuG+cflJW\njBbx8Nkt8+SCImbU6BtW/jHoIvrKOhaAtxN8532A0OEzO5CWT8xoIE4yf9Xf5lle\nC2vb3zgjlQGaTzLu9O9Si9Zc1R//Oc16w1MoDo9nz8KHfzjjTR7K426zflMFbyb4\nEr+3Ywkj+BrzFvbtexltQaZfcfF9bWsdwXbhbM7QR3R9xLfJmGnohrHS6rQ/BXjW\nsQe9x73PjbP5bguYInn+DqTvUhiYw/KbyeOEQQIDAQABAoIBAQCnqrPTLnEuLQF9\nCkhh/bnd8HCSF3VIE6tB9HQ4/yNdb404he5vEQb7JBTcBmqh1zgZPlAIAvHV6rkX\nDmZ98xXz/epeH1NjAJD05BueUPPvDO7URzeoVFE5u6RkW/jr+iAzQtAom8ZtY8Cw\nRK4eunI3cbXmeWzm6OQeHFc6q7u9cONoo3bsALsHk3rEqEH1LUj4lEQPJPOD5g2H\ntvp1vs1SdjfM5nVcyvN4u4FBom97+vPDNtl12O6TBK2viVANjkw8FOVZIRqiAiNZ\nfIqOVwacN00phMzUiQB0IH0VRl+AVpKyWn5eAHTB6KPWGjF5LQxarJ8Km/DbYG3G\n7a1gEOiBAoGBAO9jZGMfRhpGeiOOM53ccUjzrTgNWIC8x+nGN69Z/mtzn5gg8fQO\nlypQcI8LVDPojqUkkBpxi7+6IuRXpzboC3yYmyvId6EBaD0ucCsJJbZfMuKoEkwP\nX9Hif90KkaJT6CXdvayzkLsoQVlFAUvIg6h1/YeT9UHDNxIjkKwfbN4vAoGBAN6w\nkNlQegU+UaCWs99uBm5mBqkl9TsP/fTwObkvnMF9RIaMDFWi7gZIDquihp8bicWs\n9qsc//BOJH+lJzN0ychklWFiy8GvRWH0nj80CkZIdRJBru6ssv616N/pF2UvS+2k\no8vK3Etr9Q9cu/TPdFk0DLdqGWwWK9gYQRSz8RiPAoGBAJOR4cB49u4bpA9nCcq2\nqd8e2BlFoNk7hsFFv+4IvB3hGPDe3khk9irPi5OimDWnlseW0n56oHuAcyHwJtRi\nFzKnoIBNA/HsvCV7Cwp8iRLzfJrcoOriT19DES9h5IT81I8DMnnT99Rn7GDrePEO\nmpquoauCOh5gCQLVicmRVbthAoGAVCHxF6lH8GMzA7DsFCXFWEBDk/Q7Si0ojTmV\nFVnfp1pkYVDX+CKuOsFOiZnFsqb8zioip1M1ftyG/ZKv1Mjy0zrtFPX2dR564B9D\nCi3nE9acJGGcbZ/hoEmpya6OoDPWQ9pH596ki/olg8BNYpheJLV9eG4lXKijt+ix\n7dht5hECgYB++7Y6z81/+FFB4MK4gyBvwOvPhk3J80Fx5KEx7kMBZ/i6VPUuWMKJ\n0r2YQGDf2K4DsECJ2eCDizkddnLSJvXUK8XuvHY+PsAGSLr6LeO8dxRdoPe+1ciT\nM4TC9j+kmeIJanPGR6wAKQ/CMblTAUQ76ztxscvqEAOUcfa2Y1n9xQ==\n-----END RSA PRIVATE KEY-----"
```

Comment:
Move encryption keys to vars outside of the test, so that you can reuse them better.